### ISSUE TITLE
'S3.Utils.AmazonS3Uri.IsAmazonS3Endpoint' throws 'InvalidOperationException' for relative URIs

### ISSUE THREAD

**Issue by julealgon** [aws/aws-sdk-net#1783]
## Description

The utility method `IsAmazonS3Endpoint` is failing to handle relative URIs and throwing `InvalidOperationException` instead of returning `false`.

The logic inside the method checks for rules on the `Host` property of the given `Uri`, however [`Host` is not supposed to be accessed on relative urls as per documentation](https://docs.microsoft.com/en-us/dotnet/api/system.uri.host?view=net-5.0#exceptions).
> Exceptions
InvalidOperationException
This instance represents a relative URI, and this property is valid only for absolute URIs.

## Reproduction Steps

```csharp
IsAmazonS3Endpoint(new Uri("server/", UriKind.Relative));
```

### Logs

```
 System.InvalidOperationException: This operation is not supported for a relative URI.
      Stack Trace: 
        Uri.get_Host()
        AmazonS3Uri.IsAmazonS3Endpoint(Uri uri)
```

### Environment

* Package Version: AWSSDK.S3 v3.5.7.11
* OS Info:  Windows 10
* Build Environment  Visual Studio
* Targeted .NET Platform:  NET Core 3.1 and NET 5.0

## Resolution
- [ ] :wave: I can/would-like-to implement a fix for this problem myself

The method should gracefully handle relative URIs by inspecting whether the passed in value is relative or absolute before attempting to inspect absolute-only properties. It should return `false` for relative URIs.

---

This is a :bug: bug-report

**Comment by ashishdhingra** [comment]
Hi @julealgon,

Thanks for reporting the issue. Looking at the code https://github.com/aws/aws-sdk-net/blob/133eddb00a3fbd2ab395e01c3d84ac9fe1ff38b2/sdk/src/Services/S3/Custom/Util/AmazonS3Uri.cs#L229, it appears to be a bug. I would discuss with the development team to get it prioritized.

Thanks,
Ashish

**Comment by ashishdhingra** [comment]
Fixed in AWS SDK S3 version 3.5.8.6
